tensorboard/webapp/widgets/line_chart_v2/lib/worker.ts

```diff
@@ -18,10 +18,15 @@ limitations under the License.
  *
  * This module exists to conform to internal requirements.
  *
- * @param workerResourcePath URL pathanme to the JavaScript resource served by
- *   TensorBoard. TensorBoard disallows fetching JavaScript resources from a differnt
- *   origin.
+ * @param workerJsFilename Worker JavaScript resource file name. Exact resource path
+ *   depends on context. It can be relative or absolute depending on a app and its
+ *   configuration (internally, it is absolute; i.e., served at "/<resource_name.js>").
  */
-export function getWorker(workerResourcePath: string): Worker {
-  return new Worker(workerResourcePath);
+export function getWorker(workerJsFilename: string): Worker {
+  if (workerJsFilename.includes('/')) {
```

Comment:
I see that today we are serving the script via the assets `tf_web_library` with `path = "/"`, but is there any real reason why it must be this way, rather than served from, say, "/workers/chart_worker.ts" in the future?  What is the bad case that this check helps us avoid?

Reply:
In OSS, we have to support path_prefix which we really don't internally (I guess it is still possible to use the flag internally but manual invocation of TB is not the normative case anymore).

Reply:
So calling `getWorker('/foo.js')` is an error because it wouldn't work with path_prefix.  If someone called `getWorker('foo/bar.js')`, if would be valid as long as we used a tf_web_library in OSS with `path="foo"` (I think?).  Does it make sense to change the condition to `workerJsFileName.startsWith('/')`?

Reply:
While it would be valid, there is no easy way to support even "foo/bar.js" so we are giving up and just putting a constraint that it has to be a name of the file (i.e., "bar.js" in your example) and that's it.

Reply:
Gotcha, thanks for the explanation